### PR TITLE
Bug 1303604 - Update BoGo to enable more tests that fail due to RSA-PSS

### PR DESCRIPTION
**Pull request by ttaubert**
r? @franziskuskiefer

### CODE REVIEW

File: external_tests/nss_bogo_shim/config.json [hunk 1]
```diff
@@ -32,45 +29,17 @@
...
         "WrongMessageType-TLS13-EncryptedExtensions":"nss updated/broken",
         "WrongMessageType-TLS13-CertificateRequest":"nss updated/broken",
-        "WrongMessageType-ClientFinished":"nss updated/broken",
         "WrongMessageType-TLS13-ServerCertificateVerify":"nss updated/broken",
         "WrongMessageType-TLS13-ServerCertificate":"nss updated/broken",
         "WrongMessageType-TLS13-ServerFinished":"nss updated/broken",
         "EncryptedExtensionsWithKeyShare":"nss updated/broken",
         "EmptyEncryptedExtensions":"nss updated/broken",
-        "ExtendedMasterSecret-NoToNo-Server":"nss updated/broken"
+        "TLS13-RequestContextInHandshake":"Boring fails with a decode_error when the req_ctx is non-zero during the handshake",
```
Comment (ekr) [inline external_tests/nss_bogo_shim/config.json]: This is actually a defect in NSS. We changed it and I failed to update.

**Comment by ekr** [comment]
Why is this patch OK? Don't you *also* need to disable PSS in the shim?

File: external_tests/nss_bogo_shim/config.json (hunk 1 above)
Comment (ttaubert) [inline external_tests/nss_bogo_shim/config.json]: Ok, let's fix this in a separate bug after we landed this patch?

**Comment by ttaubert** [comment]
> Why is this patch OK? Don't you also need to disable PSS in the shim?

This patch bumps us to boringssl revision "57e929f3c8c3d412639eb123382c79ff3bdc3ed3", the one that introduced RSA-PSS support for TLS 1.2. I experimented at first with disabling PSS signature schemes but found that bumping was the better option and we could disable fewer tests that way.

**Comment by ekr** [comment]
OK. LGTM.

File: external_tests/nss_bogo_shim/config.json [hunk 2]
```diff
@@ -1,14 +1,11 @@
 {
     "DisabledTests": {
-        "*HelloRetryRequest*":"HRR hasn't landed yet",
-        "SecondClientHelloWrongCurve":"HRR hasn't landed yet",
+        "SendHelloRetryRequest":"Expects CurveX25519",
+        "SendHelloRetryRequest-2":"Expects CurveX25519",
```
Comment (franziskuskiefer) [inline external_tests/nss_bogo_shim/config.json]: interesting. well, this should go away soon as well.

File: external_tests/nss_bogo_shim/config.json (hunk 1 above)
Comment (franziskuskiefer) [inline external_tests/nss_bogo_shim/config.json]: yeah, maybe adjust the comment here with the according bug number

Comment (ttaubert) [inline external_tests/nss_bogo_shim/config.json]: Done.

**Pull request closed without merging**